subprojects/docs/src/docs/userguide/dependencyTypes.adoc

```diff
@@ -0,0 +1,96 @@
+
+To add some files as a dependency for a configuration, you simply pass a <<sec:file_collections,file collection>> as a dependency:
+
+++++
+<sample id="file-dependencies" dir="userguide/artifacts/externalDependencies" title="File dependencies">
+    <sourcefile file="build.gradle" snippet="file-dependencies"/>
+</sample>
+++++
+
+File dependencies are not included in the published dependency descriptor for your project. However, file dependencies are included in transitive project dependencies within the same build. This means they cannot be used outside the current build, but they can be used with the same build.
+
+You can declare which tasks produce the files for a file dependency. You might do this when, for example, the files are generated by the build.
+
+++++
+<sample id="generatedFileDependencies" dir="userguide/artifacts/generatedFileDependencies" title="Generated file dependencies">
+    <sourcefile file="build.gradle" snippet="generated-file-dependencies"/>
+    <output args="-q list"/>
+</sample>
+++++
+
+[[sub:module_dependencies]]
+=== External module dependencies
+
+External module dependencies are the most common dependencies. They refer to a module in an external repository.
+
+++++
+<sample id="moduleDependencies" dir="userguide/artifacts/externalDependencies" title="Module dependencies">
+    <sourcefile file="build.gradle" snippet="module-dependencies"/>
+</sample>
+++++
+
+See the api:org.gradle.api.artifacts.dsl.DependencyHandler[] class in the API documentation for more examples and a complete reference.
+
+Gradle provides different notations for module dependencies. There is a string notation and a map notation. A module dependency has an API which allows further configuration. Have a look at api:org.gradle.api.artifacts.ExternalModuleDependency[] to learn all about the API. This API provides properties and configuration methods. Via the string notation you can define a subset of the properties. With the map notation you can define all properties. To have access to the complete API, either with the map or with the string notation, you can assign a single dependency to a configuration together with a closure.
+
+If you declare a module dependency, Gradle looks for a module descriptor file (`pom.xml` or `ivy.xml`) in the repositories. If such a module descriptor file exists, it is parsed and the artifacts of this module (e.g. `hibernate-3.0.5.jar`) as well as its dependencies (e.g. cglib) are downloaded. If no such module descriptor file exists, Gradle looks for a file called `hibernate-3.0.5.jar` to retrieve. In Maven, a module can have one and only one artifact. In Gradle and Ivy, a module can have multiple artifacts. Each artifact can have a different set of dependencies.
```

Comment:
This bit is very Java specific, it seems like we should either avoid that, make this a NOTE, or include information for C++ etc.

Reply:
Existing, untouched content. I had just move the content out into the glossary.

> make this a NOTE

I'll make it a note for now.

> include information for C++

Unfortunately, that's not ready yet so we can't reference it.

Reply:
Feel free to ignore for now/save my comments regarding moved content.